includes/class-wc-subscriptions-order.php

```diff
@@ -932,25 +933,66 @@ public static function add_subscriptions_to_view_order_templates( $order_id ) {
 	/**
 	 * Loads the related orders table on the view subscription page
 	 *
-	 * @since 1.0.0 - Migrated from WooCommerce Subscriptions v2.0
+	 * @since 1.0.0 Migrated from WooCommerce Subscriptions v2.0.
+	 *
+	 * @param WC_Subscription $subscription The subscription whose related orders we are interested in.
 	 */
 	public static function get_related_orders_template( $subscription ) {
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$page = (int) ( $_GET['related_orders_page'] ?? 1 );
+		// phpcs:enable WordPress.Security.NonceVerification.Recommended
+		// phpcs:enable WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		// phpcs:enable WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 
-		$subscription_orders = $subscription->get_related_orders();
+		$related_orders      = $subscription->get_paginated_related_orders( 'ids', array( 'parent', 'renewal', 'switch' ), $page, 10 );
+		$subscription_orders = $related_orders->orders;
 
 		if ( 0 !== count( $subscription_orders ) ) {
 			wc_get_template(
 				'myaccount/related-orders.php',
 				array(
 					'subscription_orders' => $subscription_orders,
 					'subscription'        => $subscription,
+					'page'                => $page,
+					'max_num_pages'       => $related_orders->max_num_pages,
 				),
 				'',
 				WC_Subscriptions_Core_Plugin::instance()->get_subscriptions_core_directory( 'templates/' )
 			);
 		}
 	}
 
+	/**
+	 * Introduced to support pagination of the related orders list (within the My Account > View Subscription
+	 * screen).
+	 *
+     * @since 7.5.0 Updated to support pagination of the related orders list.
+     *
```

Comment:
FYI, there seem to be some formatting discrepancy. Looks like spaces applied instead of tabs.

Reply:
Apologies ... not sure how that slipped through. Should be addressed in https://github.com/Automattic/woocommerce-subscriptions-core/pull/833/commits/d7d92e9ff100566e11735d4fae620514c3846ba2.

Reply:
No worries, it's hard/impossible to spot. I noticed in GitHub UI, but it looked fine in IDE until I highlighted everything :D